pkg/redis/dialer.go

```diff
@@ -17,15 +18,15 @@ var dialers = map[string]Dialer{
 	DialerTcp: dialerTcp,
 }
 
-type Dialer func(logger mon.Logger, settings *Settings) func() (net.Conn, error)
+type Dialer func(logger mon.Logger, settings *Settings) func(context.Context, string, string) (net.Conn, error)
 type SrvNamingFactory func(appId cfg.AppId, name string) string
 
 var srvNamingStrategy = func(appId cfg.AppId, name string) string {
 	return fmt.Sprintf("%s.%s.redis.%s.%s", name, appId.Application, appId.Environment, appId.Family)
 }
 
-func dialerSrv(logger mon.Logger, settings *Settings) func() (net.Conn, error) {
-	return func() (net.Conn, error) {
+func dialerSrv(logger mon.Logger, settings *Settings) func(ctx context.Context, network, addr string) (net.Conn, error) {
```

Comment:
if we have a context in the dialer now, does it make sense to call `DialContext` to actually use the context?

Reply:
can do